src/Sulu/Bundle/TrashBundle/Domain/Model/TrashItem.php

```diff
@@ -0,0 +1,225 @@
+
+    public function setResourceId(string $resourceId): TrashItemInterface
+    {
+        $this->resourceId = $resourceId;
+
+        return $this;
+    }
+
+    public function getRestoreData(): array
+    {
+        return $this->restoreData;
+    }
+
+    public function setRestoreData(array $restoreData): TrashItemInterface
+    {
+        $this->restoreData = $restoreData;
+
+        return $this;
+    }
+
+    public function getResourceTitle(?string $locale = null): string
+    {
+        return $this->getTranslation($locale)->getTitle();
+    }
+
+    public function setResourceTitle(string $resourceTitle, ?string $locale = null): TrashItemInterface
+    {
+        if (!$this->hasTranslation($locale)) {
+            $translation = new TrashItemTranslation($this, $locale, $resourceTitle);
+
+            $this->translations->add($translation);
+
+            return $this;
+        }
+
+        $translation = $this->getTranslation($locale);
+        $translation->setTitle($resourceTitle);
+
+        return $this;
+    }
+
+    public function getResourceSecurityContext(): ?string
+    {
+        return $this->resourceSecurityContext;
+    }
+
+    public function setResourceSecurityContext(?string $resourceSecurityContext): TrashItemInterface
+    {
+        $this->resourceSecurityContext = $resourceSecurityContext;
+
+        return $this;
+    }
+
+    public function getResourceSecurityObjectType(): ?string
+    {
+        return $this->resourceSecurityObjectType;
+    }
+
+    public function setResourceSecurityObjectType(?string $resourceSecurityObjectType): TrashItemInterface
+    {
+        $this->resourceSecurityObjectType = $resourceSecurityObjectType;
+
+        return $this;
+    }
+
+    public function getResourceSecurityObjectId(): ?string
+    {
+        return $this->resourceSecurityObjectId;
+    }
+
+    public function setResourceSecurityObjectId(?string $resourceSecurityObjectId): TrashItemInterface
+    {
+        $this->resourceSecurityObjectId = $resourceSecurityObjectId;
+
+        return $this;
+    }
+
+    public function getTimestamp(): \DateTimeImmutable
+    {
+        return $this->timestamp;
+    }
+
+    public function setTimestamp(\DateTimeImmutable $timestamp): TrashItemInterface
+    {
+        $this->timestamp = $timestamp;
+
+        return $this;
+    }
+
+    public function getUser(): ?UserInterface
+    {
+        return $this->user;
+    }
+
+    public function setUser(?UserInterface $user): TrashItemInterface
+    {
+        $this->user = $user;
+
+        return $this;
+    }
+
+    public function getTranslation(?string $locale = null): TrashItemTranslation
+    {
+        /** @var TrashItemTranslation|false $translation */
+        $translation = $this->translations->filter(
+            function(TrashItemTranslation $translation) use ($locale) {
+                return $translation->getLocale() === $locale;
+            }
+        )->first()
+            ?: $this->translations->first();
```

Comment:
I personally not a fan of automatic fallbacks or fallbacks which can not be disabled in model methods so I would here go with:

```suggestion
    public function getTranslation(?string $locale = null, bool $fallback = false): TrashItemTranslation
    {
        /** @var TrashItemTranslation|false $translation */
        $translation = $this->translations->filter(
            function(TrashItemTranslation $translation) use ($locale) {
                return $translation->getLocale() === $locale;
            }
        )->first();
        
        if (!$translation && $fallback) {
            $translation = $this->translations->first();
        }
```